results_dashboard.py

```diff
@@ -1,20 +1,112 @@
+from app import create_app, setup_logging
+
+import logging
 import os
+import time
 
 from make_datastore import create_datastore
 
-# Determine config settings from environment variables.
-config_name = os.getenv('FLASK_CONFIG') or 'development'
 
 if __name__ == "__main__":
+    # Argument/config parsing
+    parser = configargparse.ArgParser(
+        auto_env_var_prefix='',
+        default_config_files=[
+            '/etc/seamm/seamm.ini',
+            '~/.seamm/seamm.ini',
+        ]
+    )
+    parser.add_argument(
+        '--seamm-configfile',
+        is_config_file=True,
+        default=None,
+        help='a configuration file to override others'
+    )
+
+    # Options for the dashboard
+    parser.add_argument(
+        "--log-level",
+        default='INFO',
+        choices=[
+            'CRITICAL', 'ERROR', 'WARNING', 'INFO', 'DEBUG', 'NOTSET'
+        ],
+        type=str.upper,
+        help="the logging level for the dashboard"
+    )
+    parser.add_argument(
+        "--console-log-level",
+        default='INFO',
+        choices=[
+            'CRITICAL', 'ERROR', 'WARNING', 'INFO', 'DEBUG', 'NOTSET'
+        ],
+        type=str.upper,
+        help="the logging level for the dashboard console"
+    )
+    parser.add_argument(
+        '--mode',
+        choices=['production', 'development', 'testing'],
+        type=str.lower,
+        default='production',
+        help='the mode to run in'
+    )
+    parser.add_argument(
+        "--datastore",
+        dest="datastore",
+        default='.',
+        action="store",
+        env_var='SEAMM_DATASTORE',
+        help="The datastore (directory)."
+    )
+    parser.add_argument(
+        "--database",
+        dest="database",
+        default='seamm.db',
+        action="store",
+        env_var='SEAMM_DATABASE',
+        help="The database file."
+    )
+    parser.add_argument(
+        "--logdir",
+        dest="logdir",
+        default='%datastore%/logs',
+        action="store",
+        env_var='SEAMM_LOGDIR',
+        help="The directory for logging"
+    )
+    parser.add_argument(
+        '--initialize',
+        action='store_true',
+        help='initialize the database from scratch'
+    )
+
+    options, unknown = parser.parse_known_args()
+
+    # Setup logging
+    setup_logging(options)
+    logger = logging.getLogger()
 
-    app = create_app(config_name)
+    # And proceed. First check if the database exists
+    db_file = os.path.join(
+        os.path.expanduser(options.datastore), options.database
+    )
+    initialize = not os.path.exists(db_file)
+    
+    if options.initialize and os.path.exists(db_file):
+        os.remove(db_file)
 
-    location = app.config['SQLALCHEMY_DATABASE_URI']
-    db_path = location.replace('sqlite:///', '')
-    location = os.path.dirname(db_path)
+    app = create_app(options.mode)
 
-    with app.app_context():        
-        create_datastore(location)
+    if initialize or options.initialize:
```

Comment:
This does not guarantee that the database which is initialized is bound to the app. The DB which is used for the app is set in `config`, and this argument allows you to create a database anywhere specified, if I am understanding correctly.

Reply:
I have no switched to using SQLALCHEMY_DB_URI everywhere -- obviating the need to have the file name -- and setting it from the command line or configuration. I added a default to get it from the datastore, again command line or configuration. Unfortunately at the moment relative paths don't work. Not sure why :-(